internal/worker/worker.go

```diff
@@ -129,7 +137,32 @@ func (w *worker) UnloadKmod(ctx context.Context, cfg *kmmv1beta1.ModuleConfig) e
 		return fmt.Errorf("could not unload module %s: %v", moduleName, err)
 	}
 
-	// TODO remove firmware
+	//remove firmware files only (no directories)
+	if cfg.Modprobe.FirmwarePath != "" {
+		baseImageDir := pr.fsDir + "/" + cfg.Modprobe.FirmwarePath
```

Comment:
```suggestion
		baseImageDir := filepath.Join(pr.fsDir, cfg.Modprobe.FirmwarePath)
```

Reply:
Also I think we could find a better name for `baseImageDir`. Maybe `imageFirmwarePath`?

Reply:
Fixed